compiler/rustc_passes/src/check_attr.rs

```diff
@@ -1793,6 +1794,30 @@ impl<'tcx> CheckAttrVisitor<'tcx> {
                             );
                         }
                     }
+
+                    // if the attribute is malformed, it may return None from this
+                    // but an error will have already been emitted, so this code should just skip such attributes
+                    if let Some((
+                        _,
+                        MetaItemLit {
+                            kind: ast::LitKind::Int(literal, ast::LitIntType::Unsuffixed),
+                            ..
+                        },
+                    )) = hint.singleton_lit_list()
+                    {
+                        let val = literal.get() as u64;
+                        // ignore values greater than 2^29, a different error will be emitted
+                        if val <= 2_u64.pow(29) {
```

Comment:
Problem: please use a [`self.dcx().delayed_bug("...")`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_errors/struct.DiagCtxtHandle.html#method.delayed_bug) here to assert that an error is emitted

> This is a strange one: lets you register an error without emitting it. If compilation ends without any other errors occurring, this will be emitted as a bug. Otherwise, it will be silently dropped. I.e. “expect other errors are emitted” semantics. Useful on code paths that should only be reached when compiling erroneous code.
>
> -- <https://doc.rust-lang.org/nightly/nightly-rustc/rustc_errors/enum.Level.html#variant.DelayedBug>

Reply:
Turns out this is good, actually. The delayed bug caught the fact that we didn't even properly check the attr on certain targets in the first place, lmao.